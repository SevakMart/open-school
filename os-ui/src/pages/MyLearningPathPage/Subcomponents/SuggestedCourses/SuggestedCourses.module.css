.mainContainer {
  margin: 156px 88px 130px;
  box-sizing: border-box;
}
.mainContainer .mainTitle {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #5E617B;
  margin-top: 0;
}
.mainContainer .noSuggestedCourses {
  text-align: center;
}
.mainContainer .suggestedCoursesContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 0 32px;
  justify-content: center;
}/*# sourceMappingURL=SuggestedCourses.module.css.map */