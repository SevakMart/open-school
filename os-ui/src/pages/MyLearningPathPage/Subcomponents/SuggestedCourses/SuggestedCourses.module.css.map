{"version":3,"sources":["SuggestedCourses.module.scss","SuggestedCourses.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","../../../../styles/Mixins/GridMixin.scss"],"names":[],"mappings":"AAKA;EACI,wBAAA;EACA,sBAAA;ACJJ;ADKI;EEPF,yBCDY;EDEZ,kBFOuC;EENvC,gBFM8C;EEL9C,eFKkD;EEJlD,iBFIuD;EACjD,cIRU;EJSV,aAAA;ACCR;ADCI;EACI,kBAAA;ACCR;ADCI;EKfA,aAAA;EACA,qCLeuB;EKdvB,WAAA;EACA,uBLaqC;ACIzC","file":"SuggestedCourses.module.css"}