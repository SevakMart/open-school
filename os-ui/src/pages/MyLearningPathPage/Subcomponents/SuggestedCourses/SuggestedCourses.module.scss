@import "../../../../styles/Configs/Colors.scss";
@import "../../../../styles/Configs/Fonts.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer {
    margin:156px 88px 130px;
    box-sizing: border-box;
    .mainTitle {
        @include fontStyle($primary-font,normal,500,16px,24px);
        color:$main-text-color;
        margin-top: 0;
    }
    .noSuggestedCourses {
        text-align: center;
    }
    .suggestedCoursesContainer {
        display:grid;
        grid-template-columns: repeat(4,1fr);
        justify-content:center;
        column-gap: 32px;

    }
}