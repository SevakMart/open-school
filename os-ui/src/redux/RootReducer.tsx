import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
