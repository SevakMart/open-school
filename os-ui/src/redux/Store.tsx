import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
