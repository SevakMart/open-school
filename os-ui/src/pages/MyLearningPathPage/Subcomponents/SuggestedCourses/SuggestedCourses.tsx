import { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../../../redux/Store';
import Loader from '../../../../component/Loader/Loader';
import LearningPath from '../../../../component/LearningPath/LearningPath';
import { ErrorField } from '../../../../component/ErrorField/ErrorField';
import { getSuggestedCourses } from '../../../../redux/Slices/SuggestedCourseSlice';
import styles from './SuggestedCourses.module.scss';

/* eslint-disable max-len */

const SuggestedCourses = ({ userId, token }:{userId:number, token:string}) => {
  const dispatch = useDispatch();
  const { t } = useTranslation();
  const suggestedCourseState = useSelector<RootState>((state) => state.suggestedCourse);
  const { entity, isLoading, errorMessage } = suggestedCourseState as {entity:any[], isLoading:boolean, errorMessage:string};
  const { mainContainer, suggestedCoursesContainer } = styles;

  useEffect(() => {
    dispatch(getSuggestedCourses({ userId, token }));
  }, []);
  if (isLoading) {
    return (
      <div className={mainContainer}>
        <Loader />
      </div>
    );
  }
  if (errorMessage) {
    return (
      <div className={mainContainer}>
        <ErrorField.MainErrorField className={['suggestedCourseErrorStyle']}>
          {errorMessage}
        </ErrorField.MainErrorField>
      </div>
    );
  }
  if (!entity.length) {
    return (
      <div className={mainContainer}>
        <h2>{t('No courses.suggested')}</h2>
      </div>
    );
  }
  return (
    <div className={suggestedCoursesContainer}>
      {entity.map((suggestedCourse, index) => (
        <LearningPath
          key={index}
          courseInfo={suggestedCourse}
        />
      ))}
    </div>
  );
};
export default SuggestedCourses;
