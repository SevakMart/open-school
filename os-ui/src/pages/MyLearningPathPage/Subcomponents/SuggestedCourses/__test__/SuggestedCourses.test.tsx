import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import SuggestedCourses from '../SuggestedCourses';

describe('Create several unit tests for SuggestedCourses Component', () => {
  test('Create a Snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SuggestedCourses
          userId={1}
          token="123"
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
