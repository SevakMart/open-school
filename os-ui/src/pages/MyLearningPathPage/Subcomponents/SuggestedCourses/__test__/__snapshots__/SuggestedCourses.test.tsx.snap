// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create several unit tests for SuggestedCourses Component Create a Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <p
      class="mainTitle"
    >
      string.myLearningPaths.suggestedLearningPaths
    </p>
    <div
      class="noSuggestedCourses"
    >
      <div
        class="mainContent"
      >
        <div
          class="loader"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
